### DISCUSSION TITLE
Is there any plans to support Viewport3D ans other 3D graphing features?

### DISCUSSION THREAD

**Discussion by laolarou726** [AvaloniaUI/Avalonia#8512] [Q&A]
I'm trying to port my WPF application to Avalonia. But I didn't find any control which is similar to Viewport3D. Is there any solutions for that?

**Comment by timunie** [comment]
There is an OpenGL control, but I have not much experience with it. You can see it in the demo App.

**Reply by laolarou726** [reply]
Could you please give me the link to the demos? Thank you so much!

**Reply by laolarou726** [reply]
Never mind. I just found it. Thank you!

**Reply by BAndysc** [reply]
Keep in mind the open gl control currently doesn't really work well on windows. It requires using WGL which usually make the avalonia freeze after few seconds. There was this PR https://github.com/AvaloniaUI/Avalonia/pull/6391 for easy using OpenTK in Avalonia, I've been testing it, in fact I am still using a build from this PR with a great success, but definitely I don't recommend you doing the same (relying on some inactive PR). Some time ago [kekekeks](https://github.com/kekekeks) mentioned that he is still going to [finish this before 11.0 release](https://github.com/AvaloniaUI/Avalonia/issues/5562). There was also [a PR for vulkan bindings](https://github.com/AvaloniaUI/Avalonia/pull/7753), but this was blocked by another PR, which has just been merged yesterday, so hopefully in upcoming days the 3d rendering situation will be much better and easier to handle

**Reply by maxkatz6** [reply]
Also master branch avalonia now supports 3d transformations, which is enough for some tasks. Though it would be pretty hard to build complex scenes this way.

**Reply by laolarou726** [reply]
Got it! Thank you guys so much.
